app/Http/Controllers/AuthController.php

```diff
@@ -69,16 +69,25 @@ public function refreshToken(RefreshTokenRequest $request, JWTAuth $auth): JsonR
         $remember = $request->input('remember', false);
 
         try {
-            $token = $auth->parseToken()->refresh();
-            $tokenCookie = $this->makeAuthorizationTokenCookie($token, $remember);
+            $refreshedOldToken = $auth->parseToken()->refresh();
```

Comment:
@EVikulov Thank you very much for taking the time to provide your review and suggestion. I appreciate your input and consideration. However, I respectfully disagree with this suggestion because:

1. before creating a new token, we definitely need to make sure that the old one can be refreshed.
2. claims can be easily changed. 'sub' contains the user id and the attacker may change it. 
3. this requires the implementation of the service, and the implementation of the service may be different from the one that requires JWT Auth.
4. we must invalidate the old token to make sure it can't be used later.
5. `$auth->authenticate()` does almost the same thing that you suggest on lines 73-74, but this method is abstracted from the application implementation.

Additionally, I would like to clarify that I'm unsure about the unpredictable behavior you mentioned. The code has been developed in a way that separates it into four clear stages, each with specific inputs and outputs:

1. parse the current token (input) and refresh it (output) to make sure that it can be refreshed
2. authenticate a user (output) using the refreshed token (input)
3. invalidate (output) refreshed token (input) to make sure it can't be used later
4. create a new token (output) for a user (input)

Therefore, the behavior should be predictable and consistent. If you have any specific concerns or observations regarding the code, I would be more than happy to address them and work towards a resolution.

@DenTray if you have any further insights or feedback, it would be great to know your thoughts on this.

Reply:
@astorozhevsky @EVikulov thanks for detailed description, I suppose that the internal JWT logic will be more stable that the manual decoding token, so I agree with @astorozhevsky